modules/instrument_manager/php/instrument_manager.class.inc

```diff
@@ -44,76 +44,145 @@ class Instrument_Manager extends \NDB_Menu_Filter
+                    . "been uploaded. Please contact your administrator to "
+                    . "install the instrument.";
+                http_response_code(409);
+                return;
+            }
+
+            move_uploaded_file($_FILES['install_file']['tmp_name'], $file_path);
+            chmod($file_path, 0644);
+            // Scripts in tools/ often make relative imports, so we must change
+            // our effective directory in order to use them.
+            chdir($this->path . "/tools");
+
+            // Use tools/ script to generate an SQL patch file based on the
+            // structure of the uploaded .linst file.
+            // If no quatUser is configured, automatic installation is not
+            // possible, so this is the last step.
+            $db_config = $config->getSetting('database');
+            exec(
+                'php generate_tables_sql_and_testNames.php < '
+                . escapeshellarg($file_path)
+            );
+            if (!$quat_user_is_configured) {
+                http_response_code(201);
+                return;
+            }
+
+            // Install the instrument by directly sourcing the SQL file
+            // generated by `generate_tables_sql_and_testNames.php` using bash.
+            // If installation is successful, `exec` will complete
+            // silently. Otherwise, it will return the exit code and error
+            // messsage from MySQL. This will be stored in $result and
+            // logged via LorisException.
+            $instrument = \NDB_BVL_Instrument::factory(
+                $instname,
+                '',
+                ''
+            );
+            exec(
+                "mysql".
+                " -h" . escapeshellarg($db_config['host']).
+                " -u" . escapeshellarg($db_config['quatUser']).
+                " -p" . escapeshellarg($db_config['quatPassword']).
+                " " . escapeshellarg($db_config['database']).
+                " < " . $this->path . "project/tables_sql/".
+                escapeshellarg($instrument->table . '.sql'),
+                $output, // $output and $status are created automatically
+                $status  // by `exec` and so need not be declared above.
+            );
+            // In MSQL, an exit code of 0 is a success and 1 means failure
+            if ($status) {
+                $this->tpl_data['feedback'] = "The installation of "
+                    . "$instrument->table.sql failed. "
+                    . "Please contact your administrator.";
+                error_log(
+                    "The installation of $instrument->table.sql failed. "
+                    . "Either: the instrument table exists (but is not in the "
+                    . "test_names table), or "
+                    . "LORIS could not connect to the database using the "
+                    . "credentials supplied in project/config.xml. "
+                    . print_r($object)
```

Comment:
there are many places that the config file can be other than the project directory (in fact, it can be anywhere since it can be set with an environment variable.)